types/styled-components/index.d.ts

```diff
@@ -310,10 +323,16 @@ export type BaseWithThemeFnInterface<T extends object> = <
     WithOptionalTheme<React.ComponentPropsWithRef<C>, T>
 >;
 export type WithThemeFnInterface<T extends object> = BaseWithThemeFnInterface<
-    Extract<keyof T, string> extends never ? any : T
+    AnyIfEmpty<T>
 >;
 export const withTheme: WithThemeFnInterface<DefaultTheme>;
 
+/**
+ * This interface can be augmented by users to add types to `styled-components`' default theme
+ * without needing to reexport `ThemedStyledComponentsModule`.
+ */
+// Unfortunately, there is no way to write tests for this
+// as any augmentation will break the tests for the default case (not augmented).
 // tslint:disable-next-line:no-empty-interface
 export interface DefaultTheme {}
```

Comment:
I couldn't find any way to test this in an isolated context that wouldn't affect the type of `DefaultTheme` as seen from `index.tsx`. It's very important for the existing tests that `DefaultTheme` be empty...

However, augmenting `DefaultTheme` did cause expected type errors so it seems to work 🤷‍♀️